FlagsmithClient/src/test/java/com/flagsmith/TraitsTests.kt

```diff
@@ -96,20 +96,46 @@ class TraitsTests {
         mockServer.mockResponseFor(MockEndpoint.SET_TRAITS)
         runBlocking {
             val result =
-                flagsmith.setTraitsSync(listOf(Trait(key = "set-from-client", value = "12345")), "person")
+                flagsmith.setTraitsSync(listOf(Trait(key = "set-from-client", traitValue = "12345")), "person")
             assertTrue(result.isSuccess)
             assertEquals("set-from-client", result.getOrThrow().first().key)
             assertEquals("12345", result.getOrThrow().first().stringValue)
             assertEquals("person", result.getOrThrow().first().identity.identifier)
         }
     }
 
+    @Test
+    fun testSetTraitsWithTransient() {
+        mockServer.mockResponseFor(MockEndpoint.SET_TRANSIENT_TRAITS)
+        runBlocking {
+            val result =
+                flagsmith.setTraitsSync(
+                    listOf(
+                        Trait(
+                            key = "trait-one-with-transient",
+                            traitValue = "transient-trait-one",
+                            transient = true
+                        ),
+                        Trait(
+                            key = "trait-two-with-transient",
+                            traitValue = "transient-trait-two",
+                            transient= false
+                        ),
+                        ), "identity-with-transient-traits")
+            assertTrue(result.isSuccess)
+            assertEquals("trait-one-with-transient", result.getOrThrow().first().key)
+            assertEquals("transient-trait-one", result.getOrThrow().first().stringValue)
+            assertEquals("identity-with-transient-traits", result.getOrThrow().first().identity.identifier)
+            assertTrue(result.getOrThrow().first().transient)
```

Comment:
Let's add `assertFalse(result.getOrThrow().last().transient)` (or however we can get the second item in the list).